lib/secure_headers/headers/content_security_policy.rb

```diff
@@ -147,6 +148,27 @@ def reject_all_values_if_none(source_list)
       end
     end
 
+    def normalize_uri_paths(source_list)
+      source_list.map do |source|
+        # Normalize domains ending in a single / as without omitting the slash accomplishes the same.
+        # https://www.w3.org/TR/CSP3/#match-paths § 6.6.2.10 Step 2
+        begin
+          uri = URI(source)
+          if uri.path == "/"
+            next source.chomp("/")
+          end
+        rescue URI::InvalidURIError
```

Comment:
@keithamus I know you wrote this code several years ago, but do you remember what edge case we are looking catch here?

It seems like we might be able to get away with something like:

```
source_list.map do |source|
    source.chomp("/")
end
```

I believe `.chomp` will either remove the last char if it exists or return the original string unmodified.

Reply:
I don’t think that’s sufficient as it shouldn’t always remove the trailing `/` - only when the entire path is just `/`. For example “example.com/foo/“ should remain as is, but “example.com/“ should be trimmed so it becomes “example.com”. 

The logic could be described as: if the entire contents of the URL after the domain name are “/“ then that is the equivalent of a URL with no path (so the contents after the domain name are “”).